packages/modules/neo4j/src/neo4j-container.ts

```diff
@@ -44,10 +56,29 @@ export class Neo4jContainer extends GenericContainer {
         NEO4J_apoc_ttl_schedule: this.ttl.toString(),
       });
     }
-    if (this.apoc) {
+
+    if (this.plugins.length > 0) {
+      const whitelists: string[] = [];
+
+      this.plugins.forEach((plugin) => {
+        if (plugin === Neo4jPlugin.APOC || plugin === Neo4jPlugin.APOC_EXTENDED) {
```

Comment:
We could simplify/reduce the code a bit if we combined the enum with the whitelist values. Something like:

```ts
export const PLUGINS = {
  APOC: { whitelist: "apoc.*" },
  APOC_EXTENDED: { whitelist: "apoc.*" },
  BLOOM: { whitelist: "bloom.*" },
  GRAPH_DATA_SCIENCE: { whitelist: "gds.*" },
} as const;

type Neo4jPlugin = keyof typeof PLUGINS;
```

That way it'll be easier to add new plugins. This code can also then be refactored like:

```ts
const whitelists = this.plugins.reduce((acc, plugin) => {
  const entry = plugins[plugin];
  if (entry?.whitelist) {
    acc.push(entry.whitelist);
  }
  return acc;
}, []);
```

Reply:
I agree that this should be made a bit easier to maintain.

While your approach seems good, I opted to keep the plugins & whitelist separate. Whitelists are promoted to top level record, looping of them is done with combination of map, filter and join. Should allow people to configure whitelists without touching the methods.

Reply:
> Whitelists are promoted to top level record, looping of them is done with combination of map, filter and join. Should allow people to configure whitelists without touching the methods.

Sorry I didn't understand this. When I say "that way it'll be easier to add new plugins", I mean for someone to contribute a new plugin to this code. They just add a new line to the `PLUGINS`, instead of scrolling down and adding another `if` statement.

Reply:
That is quite close to how I meant it. Instead of it being one variable (combination of plugin name and a possible whitelist) it is just two different declarations next to each other. Meaning this part:

```
export enum Neo4jPlugin {
  APOC = "apoc",
  APOC_EXTENDED = "apoc-extended",
  BLOOM = "bloom",
  GEN_AI = "genai",
  GRAPHQL = "graphql",
  GRAPH_ALGORITHMS = "graph-algorithms",
  GRAPH_DATA_SCIENCE = "graph-data-science",
  NEO_SEMANTICS = "n10s",
}

const pluginWhitelists: Partial<Record<Neo4jPlugin, string>> = {
  [Neo4jPlugin.APOC]: "apoc.*",
  [Neo4jPlugin.APOC_EXTENDED]: "apoc.*",
  [Neo4jPlugin.BLOOM]: "bloom.*",
  [Neo4jPlugin.GRAPH_DATA_SCIENCE]: "gds.*",
};
```

So there is no need to touch `if` statements.